gridt/src/app/core/api.service.spec.ts

```diff
@@ -253,6 +266,72 @@ describe("ApiService", () => {
       `${service.URL}/movements/${movement_id}/leader/1`,
       {},
       default_headers
-    )
+    );
+  });
+
+  it('should fail to read identity from server when not logged in', () => {
+    authServiceStub.readyAuthentication$ = cold('#', null , "Can't authenticate: no credentials");
+    service = new ApiService(httpClientStub, authServiceStub);
+
+    expect(
+      service.Identity$()
+    ).toBeObservable(cold("#", {}, "Can't authenticate: no credentials"));
   });
+
+  it('should fail to update bio when not logged in', () => {
+    authServiceStub.readyAuthentication$ = cold('#', null , "Can't authenticate: no credentials");
+    service = new ApiService(httpClientStub, authServiceStub);
+    let mock_bio = "My very first bio."
+    expect(service.changeBio$( mock_bio )).toBeTruthy( cold('#', null, "Can't authenticate: no credentials"))
+  })
+
+  it('should fail to update password when not logged in', () => {
+    authServiceStub.readyAuthentication$ = cold('#', null , "Can't authenticate: no credentials");
+    service = new ApiService(httpClientStub, authServiceStub);
+
+    let old_password = "ABCDEF";
+    let new_password = "abcdef";
+
+    expect(service.changePassword$( old_password, new_password )).toBeTruthy( cold('#', null, "Can't authenticate: no credentials"))
+  })
+
+  it('should be able to retreive identity', () => {
+    httpClientStub.get.and.returnValue(of(mock_identity[0]));
+    service = new ApiService(httpClientStub, authServiceStub);
+
+    expect(service.Identity$()).toBeObservable(cold('(a|)',{a: mock_identity[0]}));
+    expect(httpClientStub.get).toHaveBeenCalledWith(
+      `${service.URL}/identity`,
+      default_headers
+    );
+  })
+
+  it('should be able to update the user bio', () => {
+    httpClientStub.put.and.returnValue(of(
+      { "message": "Succesfully updated bio." }
+    ));
+
+    service = new ApiService(httpClientStub, authServiceStub);
```

Comment:
Please use dependency injection, I know this can be a quite the pain, but I think it is not honest to circumvent it like this, because there could be things going wrong in this section of the code (e.g. incorrectly configured modules) that would not be caught this way. You can create a new test suite or (and I think this is more beautiful) reset the TestBed using `TestBed.resetTestingModule(); ` that does not solve everything, but it helps.